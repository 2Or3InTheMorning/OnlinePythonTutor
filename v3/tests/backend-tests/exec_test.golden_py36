{
  "code": "exec(\"import os; os.system('echo security breach')\")\n",
  "trace": [
    {
      "line": 1,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {},
      "ordered_globals": [],
      "stack_to_render": [],
      "heap": {},
      "stdout": ""
    },
    {
      "line": 1,
      "event": "exception",
      "func_name": "<module>",
      "globals": {},
      "ordered_globals": [],
      "stack_to_render": [],
      "heap": {},
      "stdout": "",
      "exception_msg": "Exception: 'exec' is not supported by Python Tutor."
    }
  ]
}
