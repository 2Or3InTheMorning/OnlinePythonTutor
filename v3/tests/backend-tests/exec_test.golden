{
  "code": "exec(\"import os; os.system('echo security breach')\")\n", 
  "trace": [
    {
      "ordered_globals": [], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {}, 
      "heap": {}, 
      "line": 1, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {}, 
      "heap": {}, 
      "line": 1, 
      "event": "call"
    }, 
    {
      "ordered_globals": [], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {}, 
      "heap": {}, 
      "line": 1, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [], 
      "stdout": "", 
      "exception_msg": "ImportError: os not supported\nOnly these modules can be imported into Python Tutor:\n  StringIO, __future__, abc, array, bisect, cPickle,\n  calendar, cmath, collections, copy, ctypes, datetime,\n  decimal, doctest, fractions, functools, glob, hashlib,\n  heapq, inspect, io, itertools, json, locale,\n  math, operator, pickle, pprint, random, re,\n  sqlite3, string, struct, time, types, typing,\n  unittest", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {}, 
      "heap": {}, 
      "line": 1, 
      "event": "exception"
    }, 
    {
      "ordered_globals": [], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {}, 
      "heap": {}, 
      "line": 1, 
      "event": "return"
    }, 
    {
      "ordered_globals": [], 
      "stdout": "", 
      "exception_msg": "ImportError: os not supported\nOnly these modules can be imported into Python Tutor:\n  StringIO, __future__, abc, array, bisect, cPickle,\n  calendar, cmath, collections, copy, ctypes, datetime,\n  decimal, doctest, fractions, functools, glob, hashlib,\n  heapq, inspect, io, itertools, json, locale,\n  math, operator, pickle, pprint, random, re,\n  sqlite3, string, struct, time, types, typing,\n  unittest", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {}, 
      "heap": {}, 
      "line": 1, 
      "event": "exception"
    }
  ]
}
